### PR TITLE
Final round of fixes following the sha256 debugging session

### PR DESCRIPTION
**Pull request by guipublic**
Most of the fixes are about bitwise operations and constant folding

### CODE REVIEW

File: crates/acir/src/native_types/arithmetic.rs [hunk 1]
```diff
@@ -34,13 +34,45 @@ impl Default for Arithmetic {
...
+
 impl Arithmetic {
     pub const fn can_defer_constraint(&self) -> bool {
         false
     }
     pub fn num_mul_terms(&self) -> usize {
         self.mul_terms.len()
     }
+
+    pub fn get_value(
```
Comment (kevaundray) [inline crates/acir/src/native_types/arithmetic.rs:55]: This method seems to be misplaced as in ACIR, we are not concerned about the actual value that the witnesses hold

**Comment by kevaundray** [comment]
Can you split this up into seperate PRs, seems that there are a bunch of unrelated changes, that can be reviewed separately

**Comment by jfecher** [comment]
Thank you! Should we close this one now?

**Comment by guipublic** [comment]
Closing because it has been split into several PRs

**Pull request closed without merging**